orbit/pkg/table/tcc_access/tcc_access.go

```diff
@@ -0,0 +1,220 @@
+//go:build darwin
+// +build darwin
+
+package tcc_access
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+var (
+	testContext   = false
+	tccPathPrefix = ""
+	tccPathSuffix = "/Library/Application Support/com.apple.TCC/TCC.db"
+	dbQuery       = "SELECT service, client, client_type, auth_value, auth_reason, last_modified, policy_id, indirect_object_identifier, indirect_object_identifier_type FROM access;"
+	sqlite3Path   = "/usr/bin/sqlite3"
+	dbColNames    = []string{"service", "client", "client_type", "auth_value", "auth_reason", "last_modified", "policy_id", "indirect_object_identifier", "indirect_object_identifier_type"}
+)
+
+// Columns is the schema of the table.
+func Columns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		// added here
+		table.TextColumn("source"),
+		table.IntegerColumn("uid"),
+		// derived from a TCC.db
+		table.TextColumn("service"),
+		table.TextColumn("client"),
+		table.IntegerColumn("client_type"),
+		table.IntegerColumn("auth_value"),
+		table.IntegerColumn("auth_reason"),
+		table.BigIntColumn("last_modified"),
+		table.IntegerColumn("policy_id"),
+		table.TextColumn("indirect_object_identifier"),
+		table.IntegerColumn("indirect_object_identifier_type"),
+	}
+}
+
+// Generate is called to return the results for the table at query time.
+// Constraints for generating can be retrieved from the queryContext.
+
+func Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	// get all usernames on the mac
+	cmd := exec.Command("dscl", ".", "list", "/Users")
```

Comment:
saw this by accident the other day and it reminded me of you @jacobshandling, not requesting any changes just sharing in case it helps to make less system calls.

if you pass `UniqueID`, you get the uuid of each user in the same output

```
dscl . -list /Users UniqueID
```

Reply:
Nice, thank you!